api/v3/Case/Getfiles.php

```diff
@@ -141,17 +140,27 @@ function _civicrm_api3_case_getfiles_select(array $params) {
   $select = CRM_Utils_SQL_Select::from('civicrm_case_activity caseact')
     ->strict()
     ->join('ef', 'INNER JOIN civicrm_entity_file ef ON (ef.entity_table = "civicrm_activity" AND ef.entity_id = caseact.activity_id) ')
+    ->join('et', 'LEFT JOIN civicrm_entity_tag et ON (et.entity_table = "civicrm_activity" AND et.entity_id = caseact.activity_id) ')
     ->join('f', 'INNER JOIN civicrm_file f ON ef.file_id = f.id')
     ->select('caseact.case_id as case_id, caseact.activity_id as activity_id, f.id as id, act.activity_date_time')
     ->distinct();
 
   if (isset($params['case_id'])) {
-    // Isn't there some helper which will let us do more advanced
-    // SQL with $params['case_id']?
     $select->where('caseact.case_id = #caseIDs', [
       'caseIDs' => $params['case_id'],
-    ]
-    );
+    ]);
+  }
+
+  if (isset($params['tag_id'])) {
+    $tagIdParam = $params['tag_id'];
+
+    if (!is_array($tagIdParam)) {
+      $tagIdParam = ['=' => $tagIdParam];
+    }
+
+    $tagIDSQL = CRM_Core_DAO::createSQLFilter('et.tag_id', $tagIdParam);
+
+    $select->where($tagIDSQL);
```

Comment:
I'm thinking we can just extract the logic for getting the tagID SQL to a function,

So we have `$select->where($this->getTagIdSql($params['tag_id']));

Also can we add the parameter to the spec so it shows up in API explorer

Reply:
Fixed. But as this is not a class, I could not create a function which can be called with `$this`. So created `_civicase_get_tag_id_sql()` instead.